packages/fast-layouts-react/src/utilities/breakpoint-tracker.ts

```diff
@@ -0,0 +1,142 @@
+import { canUseDOM } from "exenv-es6";
+import { throttle } from "lodash-es";
+
+export type Breakpoint = number;
+
+export type Breakpoints = Breakpoint[];
+
+export type BreakpointTrackerCallback = (breakpoint: Breakpoint) => void;
+
+export const defaultBreakpoints: Breakpoints = [0, 540, 768, 1084, 1400, 1779];
+
+/**
+ * Identifies current breakpoint based on window width
+ */
+export function identifyBreakpoint(windowWidth: number, breakpoints: Breakpoints): number {
+    for (let breakpoint: Breakpoint = breakpoints.length - 1; breakpoint >= 0; breakpoint--) {
+        if (windowWidth >= breakpoints[breakpoint]) {
+            return breakpoint;
+        }
+    }
+}
+
+export class BreakpointTracker {
+
+    /**
+     * The current breakpoint.
+     */
+    private breakpoint: number;
+
+    /**
+     * The current breakpoint.
+     */
+    private _breakpointConfig: Breakpoints = defaultBreakpoints;
+
+    /**
+     * Store if requestAnimationFrame can be used
+     */
+    private useRequestAnimationFrame: boolean;
```

Comment:
It looks like our whole browser matrix now supports rAF (https://caniuse.com/#feat=requestanimationframe) - maybe we should remove the fallback and just rely on it?

Reply:
Agreed - Pulling this out. I could see it technically being helpful, but in reality I think we stick with our current browser matrix for consistency and b/c for any open source projects, they likely aren't working in browsers that dont.